python/ray/_private/accelerators/nvidia_gpu.py

```diff
@@ -53,58 +45,29 @@ def get_current_process_visible_accelerator_ids() -> Optional[List[str]]:
-        num_gpus = 0
-        if importlib.util.find_spec("GPUtil"):
-            gpu_list = GPUtil.getGPUs()
-            num_gpus = len(gpu_list)
-        elif sys.platform.startswith("linux"):
-            proc_gpus_path = "/proc/driver/nvidia/gpus"
-            if os.path.isdir(proc_gpus_path):
-                num_gpus = len(os.listdir(proc_gpus_path))
-        elif sys.platform == "win32":
-            props = "AdapterCompatibility"
-            cmdargs = ["WMIC", "PATH", "Win32_VideoController", "GET", props]
-            lines = subprocess.check_output(cmdargs).splitlines()[1:]
-            num_gpus = len([x.rstrip() for x in lines if x.startswith(b"NVIDIA")])
-        return num_gpus
+        try:
+            pynvml.nvmlInit()
+        except pynvml.NVMLError:
+            return 0  # pynvml init failed
+        device_count = pynvml.nvmlDeviceGetCount()
+        pynvml.nvmlShutdown()
+        return device_count
 
     @staticmethod
     def get_current_node_accelerator_type() -> Optional[str]:
         try:
-            if importlib.util.find_spec("GPUtil"):
-                gpu_list = GPUtil.getGPUs()
-                if len(gpu_list) > 0:
-                    gpu_list_names = [gpu.name for gpu in gpu_list]
-                    return NvidiaGPUAcceleratorManager._gpu_name_to_accelerator_type(
-                        gpu_list_names.pop()
-                    )
-            elif sys.platform.startswith("linux"):
-                proc_gpus_path = "/proc/driver/nvidia/gpus"
-                if not os.path.isdir(proc_gpus_path):
-                    return None
-                gpu_dirs = os.listdir(proc_gpus_path)
-                if len(gpu_dirs) == 0:
-                    return None
-                gpu_info_path = f"{proc_gpus_path}/{gpu_dirs[0]}/information"
-                info_str = open(gpu_info_path).read()
-                if not info_str:
-                    return None
-                lines = info_str.split("\n")
-                full_model_name = None
-                for line in lines:
-                    split = line.split(":")
-                    if len(split) != 2:
-                        continue
-                    k, v = split
-                    if k.strip() == "Model":
-                        full_model_name = v.strip()
-                        break
-                return NvidiaGPUAcceleratorManager._gpu_name_to_accelerator_type(
-                    full_model_name
-                )
-        except Exception:
-            logger.exception("Could not parse gpu information.")
-        return None
+            pynvml.nvmlInit()
+        except pynvml.NVMLError:
+            return None  # pynvml init failed
+        device_count = pynvml.nvmlDeviceGetCount()
+        cuda_devices_names = []
+        for index in range(device_count):
+            handle = pynvml.nvmlDeviceGetHandleByIndex(index)
+            cuda_devices_names.append(pynvml.nvmlDeviceGetName(handle))
```

Comment:
Seems you just use the first cuda_devices_names, so don't need this for loop?

Reply:
I just follow the old implementation which use GPUtil, but want to confirm we just want to get the first device name then?

Reply:
From your code, you only used the first one?

Reply:
Right, I'll remove the loop. Seems like the old implementation also only return the first device name despite there are multiple gpus